### PR TITLE
Attach actions from cinaps stanza to @cinaps and @runtest

### LINKED ISSUE #2801: The cinaps stanza adding actions to @runtest isn't very flexible
Currently the cinaps stanza adds all cinaps actions to the `runtest` alias. This has several drawbacks which I believe could be fixed by attaching those to a separate `cinaps` stanza:
1. If you're using cinaps to generate non-test code, and in particular main library code that your tests depend upon, it induces a dependency cycle between library and runtest alias. While you can still run `dune runtest --auto-promote || dune runtest`, you still potentially have to go through a failing `dune runtest` if the cinaps-generated code is dramatically out of sync or has not yet been generated.
2. Cinaps-generated code could potentially differ from released code, for instance if it contains `%%VERSION%%` or any other such placeholder, meaning that if you use the `cinaps` stanza, you can't test released code through opam as the cinaps actions will fail.

Moving cinaps actions to a separate alias would allow to solve both by:
1. specifying a clear set of dependencies `runtest` -> `build` -> `cinaps --auto-promote` and therefore a succeeding sequence of operation to build and test my project
2. allowing to choose whether I want to attach the `cinaps` stanza to my build or test commands/alias.

### PR DESCRIPTION
**Pull request by NathanReb**
This attach the cinaps actions to the `@cinaps` alias and add a dependency from `@runtest` to `@cinaps`! Fixes #2801

### CODE REVIEW

File: src/dune/cinaps.ml [hunk 1]
```diff
@@ -110,4 +109,10 @@ let gen_rules sctx t ~dir ~scope =
          :: List.map cinapsed_files ~f:(fun fn ->
                 let fn = Path.build fn in
                 A.diff ~optional:true fn
-                  (Path.extend_basename fn ~suffix:".cinaps-corrected")) )))
+                  (Path.extend_basename fn ~suffix:".cinaps-corrected")) ))
+  in
+  Super_context.add_alias_action sctx ~dir ~loc:(Some loc) ~stamp:"cinaps"
```
Comment (rgrinberg) [inline src/dune/cinaps.ml]: I would suggest to move `Alias.cinaps` to this module. The reason is cinaps is just an extension and I don't think it's worth hardcoding its existence in places where it's not completely necessary.

Comment (NathanReb) [reply src/dune/cinaps.ml]: Sure, I'll move it!

Comment (NathanReb) [reply src/dune/cinaps.ml]: Should it be considered a "standard" alias then?

It seems that currently it can't be made a standard alias from outside the `Alias` module.

Comment (rgrinberg) [reply src/dune/cinaps.ml]: `standard` is something that allows the alias not to error when it's empty? is that necessary? I don't think the manually defined `cinaps` alias has that either. Let's see if @diml has a preference.

Comment (NathanReb) [reply src/dune/cinaps.ml]: I moved the alias to `Cinaps` and therefore made it a regular alias. I'm happy to move it back to `Alias` or to expose `make_standard` if we want to make it that way!

Comment (rgrinberg) [reply src/dune/cinaps.ml]: Either way works for me.

**Comment by NathanReb** [comment]
Is the test failure expected? I got it locally and thought something was maybe misconfigured on my side but it looks like an omitted promotion.

**Comment by rgrinberg** [comment]
My bad. Let me push a fix.

File: src/dune/cinaps.ml [hunk 2]
```diff
@@ -110,4 +111,10 @@ let gen_rules sctx t ~dir ~scope =
...
+    (alias ~dir) action;
+  Super_context.add_alias_action sctx ~dir ~loc:(Some loc) ~stamp:"cinaps"
+    (Alias.runtest ~dir)
+    (Build.alias (alias ~dir) >>> Build.noop)
```
Comment (diml) [inline src/dune/cinaps.ml]: This is a heavy way to add a dependency to an alias. In particular, adding an action to an alias is more expensive than adding a plan dependency. So instead, we should add a dependency on `Alias.stamp_file (alias ~dir)`.

We should also bind `alias ~dir` to a variable given that it's used several times in this function.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/dune/cinaps.ml (hunk 1 above)
Comment (diml) [reply src/dune/cinaps.ml]: Making `cinaps` standard would mean that user can call `dune build @cinaps` in a project-independent manner, i.e. without having to care whether `cinaps` is used or not. This is valuable for tests for instance, but a bit less for cinaps given that it is not very widespread. 

When you think about it, it would make sense to include cinaps validation in the linting phase. We could do so by attaching it to the `lint` alias, which is already standard.

Finally, we could also have a general syntax for allowing a recursive alias to be empty. Such as `dune build @?cinaps`.

In conclusion, for now I wouldn't make `cinaps` standard. We can always revisit this decision in the future.

File: src/dune/cinaps.ml (hunk 2 above)
Comment (NathanReb) [reply src/dune/cinaps.ml]: That's good to know, I wasn't familiar enough with these parts. Just fixed it!

Sorry for the delay

Comment (diml) [reply src/dune/cinaps.ml]: No problem :)

File: test/blackbox-tests/test-cases/cinaps/run.t [hunk 3]
```diff
@@ -14,16 +14,30 @@ Test of cinaps integration
...
+The cinaps stanza offers a promotion workflow:
 
   $ dune runtest --auto-promote
   File "test.ml", line 1, characters 0-0:
-  Files _build/default/test.ml and _build/default/test.ml.cinaps-corrected differ.
+  Error: Files _build/default/test.ml and
+  _build/default/test.ml.cinaps-corrected differ.
   Promoting _build/default/test.ml.cinaps-corrected to test.ml.
   [1]
-  $ cat test.ml
```
Comment (diml) [inline test/blackbox-tests/test-cases/cinaps/run.t]: Why delete this part? It seems useful to me to see that the promoted file is what we expect.

Comment (NathanReb) [reply test/blackbox-tests/test-cases/cinaps/run.t]: Hmm yeah you're right. That's weird I don't remember intentionally removing it. I'll bring it back!

**Comment by NathanReb** [comment]
I also removed the `Build.noop` value that I originally added since it's not used in this PR anymore.

**Review by ghost — APPROVED** [review]
Looks good!

**Comment by NathanReb** [comment]
Just rebased, I'm guessing I can merge once CI is green.

**Comment by diml** [comment]
Sounds good!
